lib/hystrix-stats.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const stream = require('stream');
+
+const hystrixFormatter = require('./hystrix-formatter');
+/*
+  This class needs a function that outputs a stream of circuit data
+
+  Will also need to be a listener on status events
+*/
+
+class HystrixStats {
+  constructor () {
+    this.circuitInstanceMap = new WeakMap();
+
+    this._readableStream = new stream.Readable({
+      objectMode: true
+    });
+
+    // Need a _read() function to satisfy the protocol
+    this._readableStream._read = () => {};
+    this._readableStream.resume();
+
+    this._hystrixStream = new stream.Transform({
+      objectMode: true
+    });
+
+    // Need a _transform function to satisfy the protocol
+    this._hystrixStream._transform = this._hystrixTransformer;
+    this._hystrixStream.resume();
+
+    this._readableStream.pipe(this._hystrixStream);
+  }
+
+  // The stats coming in should be already "Reduced" and in
+  _hystrixTransformer (stats, encoding, cb) {
+    const formattedStats = hystrixFormatter(stats);
+
+    // Need to take the stats and map them to the hystrix format
+    return cb(null, `data: ${JSON.stringify(formattedStats)}\n\n`);
+  }
+
+  // Just a regular public function to get a handle on the stream
+  getHystrixStream () {
+    return this._hystrixStream;
+  }
+
+  // This will take the stats data from the listener and push it on the stream to be transformed
+  _hystrixSnapshotListener (stats) {
+    const circuit = this.circuitInstanceMap.get(this);
+    this._readableStream.push(Object.assign({}, {name: circuit.name, closed: circuit.closed, group: circuit.group}, stats));
+  }
+
+  // Register the instance of a circuit stats to listen for a snapshot event
+  // during that event, call the hystrixSnapshotListener to do the stream stuff
+  register (circuitInstance) {
```

Comment:
currently the class is exported as a new instance,  so you wouldn't be able to pass it along since the constructor is called while you require it.  

I think i do like that better though,  having the CircuitBreakers constructor calling `new HystrixStats()` and registering then,   let me change that